### ISSUE TITLE
Auto Scaling Batch Size

### ISSUE THREAD

**Issue by mogwai** [pyannote/pyannote-audio#516]
https://pytorch-lightning.readthedocs.io/en/latest/training_tricks.html#auto-scaling-of-batch-size

**Comment by hbredin** [comment]
This is not supported with our current API:

> Due to these constraints, this features does NOT work when passing dataloaders directly to .fit().

**Comment by hbredin** [comment]
Or, maybe it is supported but we would have to find a way to link `Model.batch_size` and `Task.batch_size` properties.

**Comment by mogwai** [comment]
Is this still not possible? Can we not link the batch size somehow?

**Comment by hbredin** [comment]
I am pretty sure it is possible.

However, there is absolutely no guarantee that larger batches would lead to better performance in the end.
That is why I have not started working on this feature for now...

One simple experiment that could be tried is to train the same VAD model (e.g. on AMI dataset) with several batch sizes and compare the performance on the validation set.

**Comment by mogwai** [comment]
Keeping the validation set the same size i assume and the same files of data to make sure the experiment is fair?

**Comment by hbredin** [comment]
You should not have to worry about the validation set since 
* it does not depend on the batch size anyway
* validation now runs after every epoch automatically using data from `protocol.development`.
